### ISSUE TITLE
WhereIn with subquery doesn't seem to be working

### ISSUE THREAD

**Issue by lackdaz** [tgriesser/knex#2282]
I am trying to return rows by unique columns:
```
select timestamp, device, data 
from SensorData 
where device is not null and timestamp in 
(select max(timestamp) from SensorData group by device)
```

https://runkit.com/lackdaz/59ecc4d5bab82900124ec531

But Knex query builder doesn't return unique ids as intended. Is this is a known bug?

**Comment by elhigu** [comment]
What do you mean by "not return unique ids as intended"? You are not selecting any unique id, just timestamp, device and data. What would you consider as "intended" functionality in this case?

In the runkit example the SQL generated by `knex` did seem to match the knex query.

**Comment by lackdaz** [comment]
To elaborate: I was looking to get entire rows by unique `device`(s) based on its last known timestamp, like this: https://imgur.com/a/VX92J
I ran this SQL query in DBeaver to be sure that this was the output I wanted.

I crafted the same SQL query using Knex, as shown above. And this was the result: https://imgur.com/a/9a7ni

Which does not display unique `device` (s) but instead shows an entirely different outcome. I know why SQL query is the same but I am wondering why the output is different?
